src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/ForkScanner.java

```diff
@@ -73,6 +97,21 @@
 
     private boolean walkingFromFinish = false;
 
+    NodeType currentType = null;
+    NodeType nextType = null;
+
+    public ForkScanner() {
+
```

Comment:
no setup call at all? I would just invoke setup(null) in order to get the updated logic if it changes